src/main/java/seedu/address/model/person/DateOfBirth.java

```diff
@@ -72,10 +88,13 @@ public static DateOfBirth getEmptyDateOfBirth() {
      */
 
     //found from https://mkyong.com/java/how-to-check-if-date-is-valid-in-java/
-    public static boolean isValidDateOfBirth(String test) {
+    public static boolean isValidDateOfBirth(String test, boolean isNaAllowed) {
         if (test == null) {
             return true;
         }
+        if (test.equals("NA") & isNaAllowed) {
```

Comment:
You can also use the EMPTY_DOB constant here.
```suggestion
        if (test.equals(EMPTY_DOB) & isNaAllowed) {
```